src/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    public readonly partial struct DateTime
+    {
+        internal static readonly bool s_isLeapSecondsSupportedSystem = IsLeapSecondsSupportedSystem();
+
+        public static DateTime UtcNow
+        {
+            get
+            {
+                // following code is tuned for speed. Don't change it without running benchmark.
+                long ticks = 0;
+
+                if (s_isLeapSecondsSupportedSystem)
+                {
+                    FullSystemTime time = new FullSystemTime();
+                    GetSystemTimeWithLeapSecondsHandling(ref time);
```

Comment:
Can this argument be `out` to avoid the unnecessary upfront zero-initialization?